faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -157,17 +156,24 @@ private FaunaClient(Connection connection) {
   /**
    * Creates a session client with the user secret provided. Queries submitted to a session client will be
    * authenticated with the secret provided. A session client shares its parent's {@link Connection} instance
-   * and must be closed after used.
+   * and thus it does not need be closed after its usage. Close the parent session for freeing up any
+   * resources held by the parent session and this session client.
    *
    * @param secret user secret for the session client
    * @return a new {@link FaunaClient}
    */
   public FaunaClient newSessionClient(String secret) {
-    return new FaunaClient(connection.newSessionConnection(secret));
+    return new FaunaClient(connection.newSessionConnection(secret)) {
+      @Override
+      public void close() {
```

Comment:
We really should just drop `close()` in this hierarchy entirely, since the JDK11 client doesn't expose it. If someone does decide to pass in their own executor, then it is their responsibility to clean up those resources.

Reply:
From what I've been reading so far, the `java.net.http.HttpClient` releases the resources once is not longer referenced in the code. So, I'm not sure if the default (internal) `Executor` will be properly close if we do not so (it seems there are other internal I/O resources that need to be closed as well).

That's essentially what we're doing in the new `close` method:

https://github.com/fauna/faunadb-jvm/blob/ba7e204a15706d2dc815080e17682f51de53dbc5/faunadb-common/src/main/java/com/faunadb/common/http/JavaHttpClient.java#L20

Some further info can be found in the following thread: [Close Java HTTP Client](https://stackoverflow.com/questions/53919721/close-java-http-client).

I'm all in for removing the `close()` method and let users not to worry about handling resources if possible, but I'm not sure if it can be done at this point even by using `java.net.http.HttpClient`...

> If someone does decide to pass in their own executor, then it is their responsibility to clean up those resources.

100% agreed for this case.

Reply:
Yes, I'm familiar with that thread. According to the docs themselves, the HTTPClient is does not need to be closed. The fact that it does not implement closable is key here. The exception to this is if the user creates their own Executor, which as we've discussed, is their responsibility to manage. Given all that, I don't think we need to try to do anything special, and we can just let the GC do the work.

Reply:
I removed the extra wrapper class and just set the reference to `null` on close.
Is that good enough?

Reply:
So, similar to Arnaud's question, should we then remove the `close` method entirely? (both from the `Connection` and the Java and Scala `FaunaClient`s?

Reply:
By my best read of the HTTPClient API, I believe we can eliminate it entirely.

Reply:
I have removed the `close` feature completely https://github.com/fauna/faunadb-jvm/pull/270/commits/f2298bb2f2e2004991b23813c9a8b2b1a8b64ff2